### PR TITLE
Fix template dir empty string

### LINKED ISSUE #4271: error on repo init with init.templateDir as empty string
### Reproduction steps

First, run `git config --global init.templateDir ''`. Next, put this in
`foo.c`:

```c
#include <assert.h>
#include <git2.h>
#include <stdio.h>

int main(void) {
    git_libgit2_init();
    git_repository_init_options opts;
    assert(git_repository_init_init_options(&opts, GIT_REPOSITORY_INIT_OPTIONS_VERSION) == 0);
    opts.flags = GIT_REPOSITORY_INIT_MKDIR | GIT_REPOSITORY_INIT_EXTERNAL_TEMPLATE;
    git_repository *repo;
    if (git_repository_init_ext(&repo, "foo", &opts) != 0) {
    	const git_error *err = giterr_last();
        printf("%d - %s\n", err->klass, err->message);
    }
    git_libgit2_shutdown();
    return 0;
}
```

Then run `cc -lgit2 foo.c` and `./a.out`.

### Expected behavior

The repo `foo` is created, as if init.templateDir was pointed at an empty directory. This is the current behavior of git. For example:

```sh
$ find /var/empty
/var/empty
$ git config --global init.templateDir /var/empty
$ git init -q a
$ find a
a
a/.git
a/.git/config
a/.git/HEAD
a/.git/objects
a/.git/objects/info
a/.git/objects/pack
a/.git/refs
a/.git/refs/heads
a/.git/refs/tags
$ git config --global init.templateDir ''
$ git init -q b
$ find b
b
b/.git
b/.git/config
b/.git/HEAD
b/.git/objects
b/.git/objects/info
b/.git/objects/pack
b/.git/refs
b/.git/refs/heads
b/.git/refs/tags
```

### Actual behavior

The repo `foo` is not created:

```
$ cc -lgit2 foo.c
$ ./a.out
2 - Could not find '' to stat: No such file or directory
$ find foo
foo
foo/.git
```

### Version of libgit2 (release number or SHA1)

0.25.1

### Operating system(s) tested

macOS 10.12.5 (16F73)

### PR DESCRIPTION
**Pull request by azdavis**
Closes #4271.

This is my first time contributing to this project, so I hope I did everything correctly. I also added a test, which I verified does pass with the empty-string check I added, and fails otherwise.

### CODE REVIEW

File: src/repository.c [hunk 1]
```diff
@@ -1784,7 +1784,12 @@ static int repo_init_structure(
 			default_template = true;
 		}
 
-		if (tdir) {
+		/* if tdir was the empty string, treat it like tdir was a path to an
```
Comment (pks-t) [inline src/repository.c]: We usually avoid asymmetric comments. Instead, we start multiline comments with a sole "/*". Furthermore, the fisrt character should be upper-cased. E.g.

```
/*
 * If tdir…
 * …
 */
```

See e.g. [1] as a reference (except for C99-style "//" comments). 

[1]: http://lkml.iu.edu/hypermail/linux/kernel/1607.1/00627.html

**Comment by azdavis** [comment]
Thanks for the comments. I've made some changes.

**Comment by pks-t** [comment]
Thanks for the fixes and your PR!

**Comment by ethomson** [comment]
Welcome to libgit2!  Thanks for tackling this and congrats on getting your first libgit2 PR merged!  🎉 

success-kid-first-libgit2-pull-request-merged
